mobile/src/navigation/root/root.navigation.tsx

```diff
@@ -16,11 +19,34 @@ const screenOptions: NativeStackNavigationOptions = {
 };
 
 const Root: FC = () => {
+  const { user } = useAppSelector((state) => ({ user: state.auth.user }));
+  const dispatch = useAppDispatch();
+  const [isLoading, setIsLoading] = useState(false);
+
+  useEffect(() => {
+    setIsLoading(true);
+    dispatch(loadCurrentUser()).then(() => {
+      setIsLoading(false);
+    });
+  }, [dispatch, setIsLoading]);
+
+  if (!user && isLoading) {
+    return <Spinner />;
+  }
+
   return (
     <NativeStack.Navigator screenOptions={screenOptions}>
-      <NativeStack.Screen name={RootScreenName.SIGN_IN} component={Auth} />
-      <NativeStack.Screen name={RootScreenName.SIGN_UP} component={Auth} />
-      <NativeStack.Screen name={RootScreenName.APP} component={AppNavigation} />
+      {user ? (
```

Comment:
Merge development to your branch, bcos this part of functionality has already added